xbmc/guilib/guiinfo/VideoGUIInfo.cpp

```diff
@@ -323,18 +323,22 @@ bool CVideoGUIInfo::GetLabel(std::string& value, const CFileItem *item, int cont
         }
         break;
       case LISTITEM_PLOT:
-        if (tag->m_type != MediaTypeTvShow &&
-            tag->m_type != MediaTypeVideoCollection &&
-            tag->GetPlayCount() == 0 &&
-            !CServiceBroker::GetSettings()->GetBool(CSettings::SETTING_VIDEOLIBRARY_SHOWUNWATCHEDPLOTS))
         {
-          value = g_localizeStrings.Get(20370);
-        }
-        else
-        {
-          value = tag->m_strPlot;
+          int iShowPlot = CServiceBroker::GetSettings()->GetInt(CSettings::SETTING_VIDEOLIBRARY_SHOWUNWATCHEDPLOTS);
+          if (tag->m_type != MediaTypeTvShow &&
+              tag->m_type != MediaTypeVideoCollection &&
+              tag->GetPlayCount() == 0 &&
+              ((tag->m_type == MediaTypeMovie && (iShowPlot == 2 || iShowPlot == 3)) || // show plot for unwatched movies
+              (tag->m_type == MediaTypeEpisode && (iShowPlot == 1 || iShowPlot == 3)))) // show plot for unwatched tv show episodes
```

Comment:
Instead of using magic numbers you should define constants for the possible settings values (0...3)

Example:
```C++
// values for SETTING_VIDEOLIBRARY_SHOWUNWATCHEDPLOTS
static const int VIDEOLIBRARY_PLOTS_SHOW_ALL = 0;
static const int VIDEOLIBRARY_PLOTS_SHOW_UNWATCHED_MOVIES = 1;
static const int VIDEOLIBRARY_PLOTS_SHOW_UNWATCHED_TVSHOWEPISODES = 2;
static const int VIDEOLIBRARY_PLOTS_HIDE_ALL = 3;
```

I would place these constants in settings/Settings.h